<?xml version="1.0" encoding="UTF-8"?>
<!--A brover is essentially a combination of both a rover and a base-->
<!--rtk_type counts up from 0-->
<!--rtk_type of 0: main base-->
<!--rtk_type from 1 to n-1: brover-->
<!--rtk_type of n: rover-->
<!--rover_quantity: number of rovers receiving corrections from the local unit-->
<!--local_host: the local unit-->
<!--base_host: the unit which is sending corrections to the local unit-->
<!--rover_host: the unit to which the local unit is sending corrections-->
<launch>

	<arg name="rosbag_name" default="rosbag.bag"/>
	<arg name="GLONAS" default="0"/>
  	<arg name="GPS" default="1"/>
  	<arg name="BEIDOU" default="1"/>
  	<arg name="GALILEO" default="1"/>
  	<arg name="Surveytime" default="120"/>
  	<arg name="Surveyacc" default="50000"/>

	<group ns="base">
		<node name="f9p" pkg="ublox" type="ublox_ros">
			<param name="serial_port" value="/dev/ttyACM0"/>
			<param name="chain_level" value="0"/>
			<param name="rover_quantity" value="2"/>
			<param name="base_type" value="moving"/>
			<param name="local_host" value="localhost"/>
			<param name="local_port" value="16140"/>
			<param name="rover_host" value="localhost"/>
			<param name="rover_port" value="16145"/>
			<param name="local_host2" value="localhost"/>
			<param name="local_port2" value="16150"/>
			<param name="rover_host2" value="localhost"/>
			<param name="rover_port2" value="16155"/>
			<param name="log_filename" value="../data/arrow/onecomp/base_binary"/>
			<param name="GPS" value="$(arg GPS)"/>
      		<param name="GLONAS" value="$(arg GLONAS)"/>
      		<param name="BEIDOU" value="$(arg BEIDOU)"/>
      		<param name="GALILEO" value="$(arg GALILEO)"/>
      		<param name="Surveytime" value="$(arg Surveytime)"/>
      		<param name="Surveyacc" value="$(arg Surveyacc)"/>
		</node>
	</group>

	<group ns="rover">
		<node name="f9p" pkg="ublox" type="ublox_ros">
			<param name="serial_port" value="/dev/ttyACM1"/>
			<param name="chain_level" value="1"/>
			<param name="rover_quantity" value = "0"/>
			<param name="local_host" value="localhost"/>
			<param name="local_port" value="16145"/>
			<param name="base_host" value="localhost"/>
			<param name="base_port" value="16140"/>
			<!--param name="rover_host" value="localhost"/-->
			<!--param name="rover_port" value="16155"/-->
			 <param name="log_filename" value="../data/arrow/onecomp/rover1_binary"/>
		</node>
		<node pkg="rosbag" type="record" name="record" output="screen" args="-O ../data/arrow/onecomp/$(arg rosbag_name) -a"/>
	</group>

	<group ns="rover2">
		<node name="f9p" pkg="ublox" type="ublox_ros">
			<param name="serial_port" value="/dev/ttyACM2"/>
			<param name="chain_level" value="1"/>
			<param name="rover_quantity" value = "0"/>
			<param name="local_host" value="localhost"/>
			<param name="local_port" value="16155"/>
			<param name="base_host" value="localhost"/>
			<param name="base_port" value="16150"/>
			<param name="log_filename" value="../data/arrow/onecomp/rover2_binary"/>
		</node>
		<!--node pkg="rosbag" type="record" name="record"
		output="screen" args="-O ../data/devel/tworovernew/rover2.bag -a"/-->
	</group>
</launch>
